lib/docker2aci.go

```diff
@@ -48,41 +47,124 @@ type DockerURL struct {
 	Tag       string
 }
 
-type Compression int
+type SquashAcc struct {
+	OutWriter *tar.Writer
+	Manifests []schema.ImageManifest
+	Filelist  []string
+}
 
 const (
 	defaultIndex  = "index.docker.io"
 	defaultTag    = "latest"
-	rocketDir     = "/var/lib/rkt"
 	schemaVersion = "0.1.1"
 )
 
-const (
-	Uncompressed Compression = iota
-	Gzip
-)
+// Convert generates ACI images from docker registry URLs.
+// It takes as input a dockerURL of the form:
+//
+// 	{docker registry URL}/{image name}:{tag}
+//
+// It then gets all the layers of the requested image and converts each of
+// them to ACI.
+// If the squash flag is true, it squashes all the layers in one file and
+// places this file in outputDir; if it is false, it places every layer in its
+// own ACI in outputDir.
+// It returns the list of generated ACI paths.
+func Convert(dockerURL string, squash bool, outputDir string) ([]string, error) {
+	parsedURL := parseDockerURL(dockerURL)
+
+	repoData, err := getRepoData(parsedURL.IndexURL, parsedURL.ImageName)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error getting repository data: %v\n", err)
```

Comment:
Generally speaking, don't print to stderr for a library function like this, just return the error and let the caller handle it. (You can still wrap the err if needed to make things clearer, e.g. `return fmt.Errorf("error getting repository data: %v", err)`)